mbed_lstools/main.py

```diff
@@ -82,165 +82,164 @@ def mbed_lstools_os_info():
     except ImportError:
         logging.basicConfig()
-    if opts.debug:
+
+    args = parse_cli(sys.argv)
+
+    if args.debug:
         root_logger.setLevel(logging.DEBUG)
     else:
         root_logger.setLevel(logging.INFO)
-    logger.debug("mbed-ls ver. %s", get_mbedls_version())
-    logger.debug("host: %s",  str((mbed_lstools_os_info())))
+    logger.debug("mbed-ls ver. %s", get_version())
+    logger.debug("host: %s",  str(mbed_lstools_os_info()))
 
-    mbeds = create(skip_retarget=opts.skip_retarget,
-                   list_unmounted=opts.list_unmounted,
-                   force_mock=opts.mock_platform)
+    mbeds = create(skip_retarget=args.skip_retarget,
+                   list_unmounted=args.list_unmounted,
+                   force_mock=args.command is mock_platform)
 
     if mbeds is None:
         logger.critical('This platform is not supported! Pull requests welcome at github.com/ARMmbed/mbed-ls')
         sys.exit(-1)
 
+    ret_code = args.command(mbeds, args)
+    if not ret_code:
+        ret_code = 0
 
-    if opts.list_platforms:
-        print(mbeds.list_manufacture_ids())
-        sys.exit(0)
-
-    if opts.mock_platform:
-        if opts.mock_platform == '*':
-            if opts.json:
-                print(json.dumps(mbeds.mock_read(), indent=4))
-
-        for token in opts.mock_platform.split(','):
-            if ':' in token:
-                oper = '+' # Default
-                mid, platform_name = token.split(':')
-                if mid and mid[0] in ['+', '-']:
-                    oper = mid[0]   # Operation (character)
-                    mid = mid[1:]   # We remove operation character
-                mbeds.mock_manufacture_id(mid, platform_name, oper=oper)
-            elif token and token[0] in ['-', '!']:
-                # Operations where do not specify data after colon: --mock=-1234,-7678
-                oper = token[0]
-                mid = token[1:]
-                mbeds.mock_manufacture_id(mid, 'dummy', oper=oper)
-            else:
-                logger.error("Could not parse mock from token: '%s'", token)
-        if opts.json:
-            print(json.dumps(mbeds.mock_read(), indent=4))
-
-    elif opts.json:
-        print(json.dumps(mbeds.list_mbeds_ext(), indent=4, sort_keys=True))
-
-    elif opts.json_by_target_id:
-        print(json.dumps(mbeds.list_mbeds_by_targetid(), indent=4, sort_keys=True))
-
-    elif opts.json_platforms:
-        print(json.dumps(mbeds.list_platforms(), indent=4, sort_keys=True))
-
-    elif opts.json_platforms_ext:
-        print(json.dumps(mbeds.list_platforms_ext(), indent=4, sort_keys=True))
-
-    elif opts.version:
-        print(get_mbedls_version())
-
-    else:
-        print(mbeds.get_string(border=not opts.simple, header=not opts.simple))
-
-    logger.debug("Return code: %d", mbeds.ERRORLEVEL_FLAG)
+    logger.debug("Return code: %d", ret_code)
 
-    sys.exit(mbeds.ERRORLEVEL_FLAG)
+    sys.exit(ret_code)
```

Comment:
Do you mind removing the ERRORLEVEL_FLAG variable here as well? https://github.com/ARMmbed/mbed-ls/blob/966747a8f6e62260a5bfd5ac3f69ba1b95fd1498/mbed_lstools/lstools_base.py#L68

Reply:
Yeah, can do.